raster-foundry/raster-foundry
# Make stac exports at least kind of for humans

## Overview

This PR makes STAC exports more human friendly in several ways.

- it replaces the directories-by-ID layout with directories-by-content-indication -- the scene collection is in `layer-collection/scenes/`, the label collection is in `label-collection/labels`
- ~it removes the top-level `stac-exports` folder -- I think this is more human friendly, because humans hate navigating through deeply nested directories, but no one has really indicated one way or another what that means, so your guess is as good as mine. Anyway the directories are now slightly less deeply nested.~ it no longer does this, because getting rid of this affects both the s3 and the local paths, and the debugging cycle is too long to keep doing this
- ~it moves items in collections into an `items/` directory -- I think this is more human friendly, because while humans hate navigating through deeply nested directories, they also hate when there are like a billion files in a folder where exactly one of the files tells them what's around (i.e. collection.json), and the billions of files make it hard to find the one file they might want to look at with their inefficient human eyes.~ it was going to but link topology is _way_ too easy to break right now
- it corrects the media type of the source link on the label item, since humans hate when media types are advertised incorrectly... right?

~It also, unrelated to the issue, moves all of our RF project config to the top of the directory -- this is for three reasons.~ It no longer does this because there were complications with testing containers needing files in certain places, and the prospect of making the docker context the repo root sounded painful.

<details>
 Reason one is that every time I split my tmux pane, I'd have to enter extra commands to get where I was going if I started in the wrong place. Now there are only two places (root + app-frontend) that I ever have to go instead of three. That's _progress_. Reason 2 is that it saves like three whole keystrokes when you import the project in metals, which is nice when you have to blow away your `.metals` / `.bloop`. That's become less frequent but someday when you save 2.5 seconds of reading the screen and get to pick the first option, you'll agree that it rules. Reason three is that it means we might someday be able to set up scala steward, which has been pretty helpful elsewhere. Having recently gone through an immensely painful big upgrade, I'd like to have a robot take care of a bunch of small upgrades.
</details>

### Checklist


### Demo

run this and explore:

```bash
$ aws --profile raster-foundry s3 cp s3://rasterfoundry-development-data-us-east-1/1dd96375-2d13-4ec7-94b7-3310dc8efa50/catalog.zip ./
```

## Testing Instructions

- assemble batch
- bring up database and memcached
- `./scripts/console batch bash`
- run the same command from #5405 but without all the visual vm stuff now that that's proven (copied here for convenience):

```bash
java \
  -cp /opt/raster-foundry/jars/batch-assembly.jar \
  com.rasterfoundry.batch.Main \
  write_stac_catalog \
  1dd96375-2d13-4ec7-94b7-3310dc8efa50 # replace with your export id
```

- `mkdir stac-catalogs` if you don't have one handy
- download the catalog that resulted and unzip it inside the directory from the previous step
- `cd <export id>`
- `tree` -- look how friendly it is to you, a human
- `cat catalog.json | jq .links` -- confirm the links make sense, and follow them around the catalog without looking at the available files to make sure they make sense

Closes raster-foundry/annotate#761
